### PR TITLE
Feature/add custom registration instructions

### PR DESCRIPTION
**Pull request by jeffvg**
#### Changes proposed in this Pull Request:
- pass in custom account details
- update invite and self reg
-

### CODE REVIEW

File: login-workflow/src/screens/InviteRegistrationPager.tsx [hunk 1]
```diff
@@ -233,15 +234,15 @@ export const InviteRegistrationPager: React.FC = () => {
             customDetails
                 .slice(1)
                 //@ts-ignore
-                .filter((item: ComponentType<AccountDetailsFormProps>) => item !== null)
+                .filter((item: ComponentType<CustomRegistrationForm> | null) => item !== null)
                 //@ts-ignore
```
Comment (joebochill) [inline login-workflow/src/screens/InviteRegistrationPager.tsx:238]: You might be able to remove this ts ignore now...if not add a comment to explain why we had to leave it in:

```
// @ts-ignore there won't be any nulls after we filter them
```

File: login-workflow/src/screens/SelfRegistrationPager.tsx [hunk 2]
```diff
@@ -302,15 +303,15 @@ export const SelfRegistrationPager: React.FC = () => {
             customDetails
                 .slice(1)
                 //@ts-ignore
-                .filter((item: ComponentType<AccountDetailsFormProps>) => item !== null)
+                .filter((item: ComponentType<CustomRegistrationForm> | null) => item !== null)
                 //@ts-ignore
```
Comment (joebochill) [inline login-workflow/src/screens/SelfRegistrationPager.tsx:307]: See my other comment about the ts-ignores and if we can remove them now

File: login-workflow/example/src/components/CustomDetailsScreen.tsx [hunk 3]
```diff
@@ -43,6 +43,7 @@ export const CustomDetailsScreen: React.FC<AccountDetailsFormProps> = (props) =>
                 onKeyPress={(e): void => {
                     if (e.key === 'Enter' && currencyRef.current) currencyRef.current.focus();
                 }}
+                style={{ marginTop: '32px' }}
```
Comment (joebochill) [inline login-workflow/example/src/components/CustomDetailsScreen.tsx:46]: See my suggestion above that will eliminate the need for users to have to manually add these margins

**Review by joebochill — COMMENTED** [review]
Excellent job! 

If we want to avoid users having to remember to put a margin at the top of their first custom form (and also handle the responsive change from 32 to 24px), we can do a little refactoring of the first account details page to:
- move the AccountDetailsWrapper to wrap both the built in and custom forms
- use the predefined class to handle the dynamic margins between the two form

it would look more like:

```tsx
        {
            name: 'AccountDetails',
            pageTitle: t('REGISTRATION.STEPS.ACCOUNT_DETAILS'),
            pageBody: (
                <AccountDetailsWrapper>
                    <AccountDetailsScreen
                        ...
                    />
                    {FirstCustomPage && (
                        <div className={sharedClasses.textField}>
                            <FirstCustomPage ...  />
                        </div>
                    )}
                </AccountDetailsWrapper>
            ),
        },
```

that would also necessitate simplifying the AccountDetails component to remove the `<AccountDetailsWrapper>` there (and replace it with a React Fragment `</>`) since we moved it into the pagers instead:
```tsx
    return (
        <>
            <TextField ...  />
            <TextField ... />
        </>
    );
```

**Comment by joebochill** [comment]
Oh, and can you comment out the custom pages at checkin so a fresh clone will always start with the basic bare-bones workflow.

**Comment by jeffvg** [comment]
> Excellent job!
> 
> If we want to avoid users having to remember to put a margin at the top of their first custom form (and also handle the responsive change from 32 to 24px), we can do a little refactoring of the first account details page to:
> 
> * move the AccountDetailsWrapper to wrap both the built in and custom forms
> * use the predefined class to handle the dynamic margins between the two form
> 
> it would look more like:
> 
> ```tsx
>         {
>             name: 'AccountDetails',
>             pageTitle: t('REGISTRATION.STEPS.ACCOUNT_DETAILS'),
>             pageBody: (
>                 <AccountDetailsWrapper>
>                     <AccountDetailsScreen
>                         ...
>                     />
>                     {FirstCustomPage && (
>                         <div className={sharedClasses.textField}>
>                             <FirstCustomPage ...  />
>                         </div>
>                     )}
>                 </AccountDetailsWrapper>
>             ),
>         },
> ```
> 
> that would also necessitate simplifying the AccountDetails component to remove the `<AccountDetailsWrapper>` there (and replace it with a React Fragment `</>`) since we moved it into the pagers instead:
> 
> ```tsx
>     return (
>         <>
>             <TextField ...  />
>             <TextField ... />
>         </>
>     );
> ```

Change here? 
and what about onDetailsChanged={setAccountDetails}
                        initialDetails={accountDetails}
                        onSubmit={

image

**Review by joebochill — APPROVED** [review]

**Review by joebochill — CHANGES_REQUESTED** [review]
One last thing...you need to update the docs for registration customization (https://github.com/pxblue/react-workflows/blob/feature/add-custom-registration-instructions/login-workflow/docs/customization.md#registration-details). You can pretty much copy right from the changes in the RN workflow:

https://github.com/pxblue/react-native-workflows/blob/dev/login-workflow/docs/customization.md#registration-details

**Review by joebochill — APPROVED** [review]

**Pull request merged**